torch/csrc/jit/tracer.h

```diff
@@ -216,34 +213,35 @@ void addInputs(Node *n, const char * name, std::array<bool, N> value) {
   throw std::runtime_error("Found an unsupported argument type in the JIT tracer. File a bug report.");
 }
 
-template<typename T, typename... Args>
-void addInputs(Node *n, const char * arg_name, T arg, const char * next_arg_name, Args... args) {
-  addInputs(n, arg_name, arg);
-  addInputs(n, next_arg_name, args...);
-}
-
-} // namespace detail
+TORCH_API void postRecordTrace(Node* node, at::ArrayRef<Variable> outputs);
 
-// NB: if you change this function, you might want to take a look at
-// preRecordPythonTrace from python_tracer.cpp
-template<typename... Args>
-PreTraceInfo preRecordTrace(Symbol op, Args... inputs) {
-  PreTraceInfo info;
-  auto & state = getTracingState();
-  auto & graph = state->graph;
-
-  Node * n = info.n = graph->create(op, /*outputs=*/0);
-  recordSourceLocation(n);
-
-  detail::addInputs(n, inputs...);
+// All these overloads are to handle (1) single `at::Tensor` (2)
+// `vector<at::Tensor>` because of implicit conversion trouble with `Variable`,
+// as well as any other types, for which we throw an exception.
```

Comment:
Why was a single overload sufficient before, but we need those now?

Reply:
Because in userland we deal with `at::Tensor`, while in internal tracer land we deal with `Variable`